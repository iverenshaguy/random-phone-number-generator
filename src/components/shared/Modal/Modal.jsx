import React, { Fragment, useState } from 'react';
import { Button, Modal as ReactStrapModal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';


export default function Modal({ disabled, label, title, actionButton, children }) {
  const [show, setShow] = useState(false);

  const toggle = () => {
    setShow(!show)
  }

  return (
    <Fragment>
      <Button disabled={disabled} className="w-50 h-25 mx-auto mb-4" onClick={toggle}>{label}</Button>
      <ReactStrapModal isOpen={show} toggle={toggle}>
        <ModalHeader toggle={toggle}>{title}</ModalHeader>
        <ModalBody>
          {children}
        </ModalBody>
        <ModalFooter>
          {actionButton  }{' '}
          <Button color="danger" onClick={toggle}>Cancel</Button>
        </ModalFooter>
      </ReactStrapModal>
    </Fragment>
  )
}