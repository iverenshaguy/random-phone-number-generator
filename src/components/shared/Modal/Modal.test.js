import React from 'react';
import PropTypes from 'prop-types';

import Modal from './Modal';

describe('Modal', () => {
  const props = {
    onPageChange: jest.fn(),
    metadata: {
      page: 1,
      perPage: 10,
      totalCount: 450,
    }
  };
  const wrapper = mount(
    <Modal {...props} />, {
      context: {
        windowWidth: 1000,
        windowHeight: 1500,
        isMobileSized: false,
      },
      childContextTypes: {
        windowWidth: PropTypes.number,
        windowHeight: PropTypes.number,
        isMobileSized: PropTypes.bool,
      }
    }
  );

  it('should render Modal component correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
