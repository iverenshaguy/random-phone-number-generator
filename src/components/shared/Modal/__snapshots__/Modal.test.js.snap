// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render Modal component correctly 1`] = `
<Modal
  metadata={
    Object {
      "page": 1,
      "perPage": 10,
      "totalCount": 450,
    }
  }
  onPageChange={[MockFunction]}
>
  <Button
    className="w-50 h-25 mx-auto mb-4"
    color="secondary"
    onClick={[Function]}
    tag="button"
  >
    <button
      aria-label={null}
      className="w-50 h-25 mx-auto mb-4 btn btn-secondary"
      onClick={[Function]}
      type="button"
    />
  </Button>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    toggle={[Function]}
    unmountOnClose={true}
    zIndex={1050}
  />
</Modal>
`;
